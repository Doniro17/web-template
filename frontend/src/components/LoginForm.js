import React, { useState } from 'react';
import { useHistory } from 'react-router-dom';
import { connect } from 'react-redux';
import { setUserData } from '../store/actions';
import Input from './Input';
import API from '../utils/API';

const LoginForm = ({ dispatch }) => {
  const history = useHistory();
  const [formUser, setFormUser] = useState({
    login: '',
    password: '',
  });
  const { login, password } = formUser;
  function onChangeLogin(event) {
    const {
      target: { value },
    } = event;
    setFormUser({
      password,
      login: value,
    });
  }
  function onChangePassword(event) {
    const {
      target: { value },
    } = event;
    setFormUser({
      login,
      password: value,
    });
  }

  function onSend(event) {
    const loginToSystem = async () => {
      event.preventDefault();
      await API.post(
        'users/login',
        {
          username: login,
          password,
        },
        {
          withCredentials: true,
        },
      )
        .then((result) => {
          console.log(result.data.id);
          dispatch(
            setUserData({
              id: result.data.id,
              login,
              roles: result.data.role,
            }),
          );
          if (result.data.role.includes('client')) {
            history.push('/products');
          } else {
            history.push('/deliveries');
          }
        })
        .catch((e) => {
          if (e.response.status === 401) {
            // eslint-disable-next-line no-alert
            alert('Incorrect email or password');
          }
        });
    };
    loginToSystem();
  }
  return (
    <form onSubmit={onSend} className="login__form" data-testid="login-form">
      <h2 className="form__title">Login</h2>
      <div className="form__inputs">
        <Input
          type="text"
          value={login}
          onChange={onChangeLogin}
          placeholder="email"
          className="form__input"
          labelStyle="container"
        />
        <br />
        <Input
          type="password"
          value={password}
          onChange={onChangePassword}
          placeholder="password"
          className="form__input"
          labelStyle="container"
        />
        <br />
        <Input type="submit" value="Login" className="form__submit" />
      </div>
    </form>
  );
};

export default connect()(LoginForm);
