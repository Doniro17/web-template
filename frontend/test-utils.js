import React from 'react'; // eslint-disable-line no-unused-vars

import { initFontAwesomeLibrary } from './src/utils';

import '@testing-library/jest-dom/extend-expect';

initFontAwesomeLibrary();

// re-export everything
export * from '@testing-library/react';
