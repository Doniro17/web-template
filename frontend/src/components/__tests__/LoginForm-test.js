import { render, fireEvent } from 'test-utils';
import LoginForm from '../LoginForm';

describe('LoginForm', () => {
  it('renders correctly', () => {
    const { container } = render(<LoginForm />);
    expect(container.firstChild).toMatchSnapshot();
  });
  it('Handle Login change correctly', () => {
    const { getByPlaceholderText } = render(<LoginForm />);
    const input = getByPlaceholderText('email');
    fireEvent.change(input, { target: { value: 'test' } });
    expect(input.value).toBe('test');
  });
  it('Handle Password change correctly', () => {
    const { getByPlaceholderText } = render(<LoginForm />);
    const input = getByPlaceholderText('password');
    fireEvent.change(input, { target: { value: 'pass' } });
    expect(input.value).toBe('pass');
  });
});
