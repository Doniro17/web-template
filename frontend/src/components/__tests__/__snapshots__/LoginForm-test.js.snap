// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm renders correctly 1`] = `
<form
  class="login__form"
>
  <h2
    class="form__title"
  >
    Login
  </h2>
  <div
    class="form__inputs"
  >
    <label
      class="container"
    >
      <input
        class="form__input"
        placeholder="email"
        type="text"
        value=""
      />
    </label>
    <br />
    <label
      class="container"
    >
      <input
        class="form__input"
        placeholder="password"
        type="password"
        value=""
      />
    </label>
    <br />
    <label>
      <input
        class="form__submit"
        type="submit"
        value="Login"
      />
    </label>
  </div>
</form>
`;
